File: common/src/main/java/org/apache/gravitino/dto/responses/UserResponse.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.gravitino.dto.responses;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.base.Preconditions;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;
import org.apache.commons.lang3.StringUtils;
import org.apache.gravitino.dto.authorization.UserDTO;

/** Represents a response for a user. */
@Getter
@ToString
@EqualsAndHashCode(callSuper = true)
public class UserResponse extends BaseResponse {

  @JsonProperty("user")
  private final UserDTO user;

  /**
   * Constructor for UserResponse.
   *
   * @param user The user data transfer object.
   */
  public UserResponse(UserDTO user) {
    super(0);
    this.user = user;
  }

  /** Default constructor for UserResponse. (Used for Jackson deserialization.) */
  public UserResponse() {
    super();
    this.user = null;
  }

  /**
   * Validates the response data.
   *
   * @throws IllegalArgumentException if the name or audit is not set.
   */
  @Override
  public void validate() throws IllegalArgumentException {
    super.validate();

    Preconditions.checkArgument(user != null, "user must not be null");
    Preconditions.checkArgument(
        StringUtils.isNotBlank(user.name()), "user 'name' must not be null and empty");
    Preconditions.checkArgument(user.auditInfo() != null, "user 'auditInfo' must not be null");
  }
}

File: clients/client-python/gravitino/dto/responses/user_response.py
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.

from __future__ import annotations

from dataclasses import dataclass, field

from dataclasses_json import config, dataclass_json

from gravitino.dto.authorization.user_dto import UserDTO
from gravitino.dto.responses.base_response import BaseResponse
from gravitino.utils.precondition import Precondition


@dataclass_json
@dataclass
class UserResponse(BaseResponse):
    """Represents a response for a user."""

    _user: UserDTO = field(default=None, metadata=config(field_name="user"))

    def user(self) -> UserDTO:
        return self._user

    def validate(self) -> None:
        Precondition.check_argument(
            self._user is not None, "User response should have a user"
        )


@dataclass_json
@dataclass
class UserListResponse(BaseResponse):
    """Represents a response for a list of users with details."""

    _users: list[UserDTO] = field(
        default_factory=list, metadata=config(field_name="users")
    )

    def users(self) -> list[UserDTO]:
        return self._users

    def validate(self) -> None:
        Precondition.check_argument(
            self._users is not None, "User list response should have users"
        )


@dataclass_json
@dataclass
class UserNamesListResponse(BaseResponse):
    """Represents a response for a list of user names."""

    _names: list[str] = field(default_factory=list, metadata=config(field_name="names"))

    def names(self) -> list[str]:
        return self._names

    def validate(self) -> None:
        Precondition.check_argument(
            self._names is not None, "User names list response should have names"
        )
